### ISSUE TITLE
Truncated list items in dropdowns and menus are not accessible for non-mouse users

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#9428]
## Detailed description

Long list items can be truncated and we reveal the full names of those list item _only_ via tooltip creating an non-accessible experience for non mouse users. 

![128064503-0049ace8-fdbf-4d27-9f59-abd5b234ca59](https://user-images.githubusercontent.com/40970507/128402561-2228c7d7-0828-480f-a7a9-d1463e6bc176.png)

## Potential solution

Using an official Carbon Tooltip that throws on focus potentially

**Comment by guidari** [comment]
I just tested this with VO and it is working.

**Comment by grahamharper** [comment]
Was the potential solution identified for this issue actually implemented?

@guidari comments indicated this works with VO but screenreader accessibility was not the original issue. The issue is that the full text is not visible to a sighted user who is not using a mouse e.g. keyboard-only or touch.

**Comment by guidari** [comment]
Oh I see the issue now! Maybe this will be fix here on that one https://github.com/carbon-design-system/carbon/issues/14676. A few change are being made on the Menus. 

I'll let the issue owner of https://github.com/carbon-design-system/carbon/issues/14676  knows about this issue.
